tools/benchcomp/benchcomp/cmd_args.py

```diff
@@ -0,0 +1,180 @@
+# Copyright Kani Contributors
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+# Command line argument processing
+
+
+import argparse
+import importlib
+import pathlib
+
+import benchcomp
+import benchcomp.entry.benchcomp
+import benchcomp.entry.run
+
+
+_EPILOG = """
+benchcomp can help you to understand the difference between two or more
+toolchains, by running benchmarks that use those toolchains and comparing the
+results.
+
+benchcomp runs two or more 'variants' of a set of benchmark suites, and compares
+and visualizes the results of these variants. This allows you to understand the
+differences between the two variants, for example how they affect the
+benchmarks' performance or output or even whether they pass at all.
+
+benchmark is structured as a pipeline of several commands. Running `benchcomp`
+runs each of them sequentially. You can run the subcommands manually to dump the
+intermediate files if required.
+"""
+
+
+def _existing_directory(arg):
+    path = pathlib.Path(arg)
+    if not path.exists():
+        raise ValueError(f"directory '{arg}' must already exist")
+    return path
+
+
+def _non_existing_directory(arg):
+    path = pathlib.Path(arg)
+    if path.exists():
+        raise ValueError(f"directory '{arg}' must not already exist")
+    return path
+
+
+def _get_args_dict():
+    return {
+        "top_level": {
+            "description":
+                "Run and compare variants of a set of benchmark suites",
+            "epilog": _EPILOG,
+        },
+        "args": [],
+        "subparsers": {
+            "title": "benchcomp subcommands",
+            "description":
+                "You can invoke each stage of the benchcomp pipeline "
+                "separately if required",
+            "parsers": {
+                "run": {
+                    "help": "run all variants of all benchmark suites",
+                    "args": [{
+                        "flags": ["--out-prefix"],
+                        "metavar": "D",
+                        "type": pathlib.Path,
+                        "default": benchcomp.entry.run.get_default_out_prefix(),
+                        "help":
+                            "write suite.yaml files to a new directory under D "
+                            "(default: %(default)s)",
+                    }, {
+                        "flags": ["--out-dir"],
```

Comment:
I've just added epilogs for each subcommand that will appear in the help text, hopefully this clarifies the semantics:

```
The run command writes one YAML file for each (suite, variant) pair.
These YAML files are in "suite.yaml" format.  Typically, users should
read the combined YAML file emitted by `benchcomp collate` rather than
the multiple YAML files written by `benchcomp run`.

The `run` command writes its output files into a directory, which
`collate` then reads from. By default, `run` writes the files into a new
directory with a common prefix on each invocation, meaning that all
previous runs are preserved without the user needing to specify a
different directory each time. Benchcomp also creates a symbolic link to
the latest run. Thus, the directories after several runs will look
something like this:

  /tmp/benchcomp/suites/2F0D3DC4-0D02-4E95-B887-4759F08FA90D
  /tmp/benchcomp/suites/119F11EB-9BC0-42D8-9EC1-47DFD661AC88
  /tmp/benchcomp/suites/A3E83FE8-CD42-4118-BED3-ED89EC88BFB0
  /tmp/benchcomp/suites/latest -> /tmp/benchcomp/suites/119F11EB...

'/tmp/benchcomp/suites' is the "out-prefix"; the UUID is the "out-dir";
and 'latest' is the "out-symlink". Users can set each of these manually
by passing the corresponding flag, if needed.

Passing `--out-symlink ./latest` will place the symbolic link in the
current directory, while keeping all runs under /tmp to avoid clutter.
If you wish to keep all previous runs in a local directory, you can do
so with

    `--out-prefix ./output --out-symlink ./output/latest`
```